core/execution/future/market.go

```diff
@@ -2977,7 +2975,35 @@ func (m *Market) updateIsolatedMarginOnAggressor(ctx context.Context, pos *posit
 	if err != nil {
 		return err
 	}
-	risk, err := m.risk.UpdateIsolatedMarginOnAggressor(ctx, mpos, marketObservable, increment, orders, trades, marginFactor, order.Side, isAmend)
+
+	totalTrades := uint64(0)
+	for _, t := range trades {
+		totalTrades += t.Size
+	}
+
+	clonedOrders := make([]*types.Order, 0, len(orders))
+	found := false
+	for _, o := range orders {
+		if o.ID == order.ID {
+			clonedOrder := order.Clone()
+			clonedOrder.Remaining -= totalTrades
```

Comment:
Not sure about this, An iceberg order with a peak size of 10, and 15 reserved remaining could produce a trade of size > 10.

Reply:
I'm not the expert but I think it might be fine as we're using TrueRemaining which would factor that in? @wwestgarth

Reply:
`cloneOrder.TrueRemaining()` returns `Remaining + IcebergOrder.ReservedRemaining` for iceberg orders, but I'm not 100% on what the orders we're dealing with here look like, so it's possible that this is fine

Reply:
I think its ok, bceause `order` is aggressive and iceberg orders only get their peaks once they're passive and on the book. So at this point it won't have been iceberg-ed yet.